ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V2.hs

```diff
@@ -207,8 +210,9 @@ mkInternals bss h =
         eFrk <- newForkerAtTarget h reg VolatileTip
         case eFrk of
           Left{} -> error "Unreachable, Volatile tip MUST be in LedgerDB"
-          Right frk ->
+          Right frk -> do
             forkerPush frk st >> atomically (forkerCommit frk) >> forkerClose frk
+            getEnv h pruneLedgerSeq
```

Comment:
I don't see why we are still pruning in here? I thought we would only prune at garbage collection, which we do below. Couldn't a snapshot be started in between this line and the previous one and then we would close the handle while taking the snapshot?

Reply:
Ah wait, if we commit the forker, and we try to take a snapshot, then we will take the snapshot of the committed imm tip, which will not be pruned ever

Reply:
Still this is confusing, because we prune on every change it seems, and the title of the PR says precisely that we prune on GC only.

Reply:
Ah nevermind. This is part of the testing internals, I was missing that.